### PR TITLE
feat: Migrate DNS Providers and DNS Entries to UI5Card

### LINKED ISSUE #4850: feat: Migrate DNS Providers panels to UI5Card (pilot)
**Description**

As a first real-world test of the panel-to-card migration (see #4830, #4849), migrate the DNS Providers resource detail view from `UI5Panel` to the new `UI5Card` component.

DNS Providers is a relatively simple resource, making it a good candidate to identify any friction points in the migration process before rolling it out to all other panels.

The goal is not just to complete the migration, but to document any issues, edge cases, or patterns discovered — which will inform the remaining migration issues.

**Reasons**

Starting with a simple, well-scoped resource lets us validate the migration approach and uncover problems early, before touching more complex panels.

**Acceptance Criteria**

- [ ] DNS Providers detail view migrated from `UI5Panel` to `UI5Card`
- [ ] No visual regressions in DNS Providers detail view
- [ ] Any edge cases or migration complexity discovered are documented as comments on this issue or as follow-up notes on the parent issue #4830

**Testing Strategy (DoD)**

- automated unit test for the new logic
- automated integration test (a new test suite or extending an existing test)

### PR DESCRIPTION
**Pull request by OliwiaGowor**
**Description**

Changes proposed in this pull request:

- Migrated DNS Entries to UI5Card
- Migrated DNS Providers to UI5Card
- Migrated ResourceGraph to UI5Card

**Related issue(s)**
Closes #4850 

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - revert: Revert commit
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: src/components/Extensibility/components/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  embedResource,
+  arrayItems = [],
+  ...props
+}: CardProps) {
+  const { widgetT } = useGetTranslation();
+  const [isDecoded] = useState(false);
+
+  if (isDecoded) {
+    value = mapValues(value, base64Decode);
+  }
```
Comment (chriskari) [inline src/components/Extensibility/components/Card.tsx]: this is dead code

**Review by chriskari — DISMISSED** [review]

File: docs/contributor/extensibility/50-list-and-details-widgets.md [hunk 2]
```diff
@@ -400,6 +401,29 @@ See the following example:
 
 <img src="./assets/display-widgets/Alert.png" alt="Example of a text widget" width="70%">
 
+### `Card`
+
+The `Card` widgets render children in a card with its own title (based on its `source` or `name`). When the source resolves to an array, all items are rendered inside a single card.
```
Comment (grego952) [inline docs/contributor/extensibility/50-list-and-details-widgets.md]:
```suggestion
The `Card` widgets render children in a card with its own title, based on its `source` or `name`. When the source resolves to an array, all items are rendered inside a single card.
```

**Review by grego952 — APPROVED** [review]

**Review by chriskari — APPROVED** [review]
